### ISSUE TITLE
SonarQube 7.6 compatibility

### ISSUE THREAD

**Issue by Vampire** [checkstyle/sonar-checkstyle#191]
While plugin version 4.17 is not even listed on https://docs.sonarqube.org/display/PLUG/Plugin+Version+Matrix, it also is not compatible with SonarQube 7.6.

If you try to do an analysis, you get

```exception
Caused by: java.lang.IllegalStateException: Unable to load component class org.sonar.plugins.checkstyle.CheckstyleConfiguration
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:65)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:632)
        at org.picocontainer.parameters.BasicComponentParameter$1.resolveInstance(BasicComponentParameter.java:118)
        at org.picocontainer.parameters.ComponentParameter$1.resolveInstance(ComponentParameter.java:136)
        at org.picocontainer.injectors.SingleMemberInjector.getParameter(SingleMemberInjector.java:78)
        at org.picocontainer.injectors.ConstructorInjector$CtorAndAdapters.getParameterArguments(ConstructorInjector.java:309)
        at org.picocontainer.injectors.ConstructorInjector$1.run(ConstructorInjector.java:335)
        at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
        at org.picocontainer.injectors.ConstructorInjector.getComponentInstance(ConstructorInjector.java:364)
        at org.picocontainer.injectors.AbstractInjectionFactory$LifecycleAdapter.getComponentInstance(AbstractInjectionFactory.java:56)
        at org.picocontainer.behaviors.AbstractBehavior.getComponentInstance(AbstractBehavior.java:64)
        at org.picocontainer.behaviors.Stored.getComponentInstance(Stored.java:91)
        at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:63)
        ... 172 more
Caused by: org.picocontainer.injectors.AbstractInjector$UnsatisfiableDependenciesException:
 org.sonar.plugins.checkstyle.CheckstyleConfiguration has unsatisfied dependency 
'class org.sonar.api.profiles.RulesProfile' for constructor 
'public org.sonar.plugins.checkstyle.CheckstyleConfiguration(
org.sonar.api.config.Configuration,org.sonar.plugins.checkstyle.CheckstyleProfileExporter,
org.sonar.api.profiles.RulesProfile,org.sonar.api.batch.fs.FileSystem)'
 from org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@3df83103:116
<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@50ebd3a:465
<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@382b9fde:54
<|
        at org.picocontainer.injectors.ConstructorInjector.getGreediestSatisfiableConstructor(ConstructorInjector.java:191)
        at org.picocontainer.injectors.ConstructorInjector.getGreediestSatisfiableConstructor(ConstructorInjector.java:110)
        at org.picocontainer.injectors.ConstructorInjector.access$100(ConstructorInjector.java:51)
        at org.picocontainer.injectors.ConstructorInjector$1.run(ConstructorInjector.java:331)
        at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
        at org.picocontainer.injectors.ConstructorInjector.getComponentInstance(ConstructorInjector.java:364)
        at org.picocontainer.injectors.AbstractInjectionFactory$LifecycleAdapter.getComponentInstance(AbstractInjectionFactory.java:56)
        at org.picocontainer.behaviors.AbstractBehavior.getComponentInstance(AbstractBehavior.java:64)
        at org.picocontainer.behaviors.Stored.getComponentInstance(Stored.java:91)
        at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:63)
        ... 186 more
```

Deprecated but still present in api:
https://github.com/SonarSource/sonarqube/blob/master/sonar-plugin-api/src/main/java/org/sonar/api/profiles/RulesProfile.java

**Comment by romani** [comment]
@muhlba91 , please help us to resolve new API change.
Thanks in advance.

**Comment by Vampire** [comment]
The findbugs plugin had the same issue and fixed it quite fast, maybe you can have a look what was necessary to fix it for them

**Comment by romani** [comment]
@daniilyar , can you help us to create docker image for 7.6-community? @muhlba91  will use it in PR to automatically prove that it works.

instruction on how I did it before - https://github.com/checkstyle/sonar-checkstyle/wiki/How-to-push-Dockerfile-to-docker-hub

**Comment by romani** [comment]
fix is merged.

**Comment by CelesteGH** [comment]
Hi
Is 4.18 going to be available today ?

**Comment by romani** [comment]
It will be available after #196

**Comment by charlesfendt** [comment]
Crash...
I just open a new issue
https://github.com/checkstyle/sonar-checkstyle/issues/208
